src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/MemoryPoolViewBufferScope.cs

```diff
@@ -13,7 +13,7 @@ namespace Microsoft.AspNetCore.Mvc.ViewFeatures.Internal
     /// </summary>
     public class MemoryPoolViewBufferScope : IViewBufferScope, IDisposable
     {
-        public static readonly int SegmentSize = 512;
+        public static readonly int MinimumSize = 16;
```

Comment:
I significantly reduced the chunk size here and it had a large positive impact. We need to do more tuning, but the values in this PR are a good start.